### PR TITLE
[PUBDEV-7852] Refactor PreviewParseWriter to 'guessTypes' Method to be Used Independently

### PR DESCRIPTION
**Pull request by mn-mikke**
This PR helps to get rid of [IcedHashMapWrapper](https://github.com/h2oai/sparkling-water/blob/b2f2d55009845a4be9e5d9f0fd3d9b63140addca/extensions/src/main/scala/water/parser/IcedHashMapWrapper.java) in Sparkling Water.

### CODE REVIEW

**Review by honzasterba — APPROVED** [review]
lgtm

File: h2o-core/src/main/java/water/parser/PreviewParseWriter.java [hunk 1]
```diff
@@ -131,86 +131,102 @@ public void addNAs(int colIdx, int nrow) {
...
+    
+    // categorical or string?
+    // categorical with 0s for NAs
+    if (nzeros > 0
+        && ((nzeros + nstrings) >= nonemptyLines) //just strings and zeros for NA (thus no empty lines)
+        && (domain.size() <= 0.95 * nstrings)) { // not all unique strings
+      return Vec.T_CAT;
+    }
+    // categorical mixed with numbers
+    if (nstrings >= (nnums+nzeros) // mostly strings
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/parser/PreviewParseWriter.java:222]: Unnecessary parentheses + wrong formatting (minor issue). Feel free to keep the parentheses in place if you feel like it, please. This is only a matter of personal taste.

Comment (mn-mikke) [reply h2o-core/src/main/java/water/parser/PreviewParseWriter.java:222]: @Pscheidl It was like this before, I just changed the variable names. I will fix the parentheses and spaces around `+`, but will leave it splitted the condition into two lines due to preserving the comments.

Comment (mn-mikke) [reply h2o-core/src/main/java/water/parser/PreviewParseWriter.java]: On second thought, I've decided to leave the parentheses as they are. They are used like this on more places within this class. If the usage is inappropriate, we could fix it in a different PR, just dedicated to this problem.

**Pull request merged**